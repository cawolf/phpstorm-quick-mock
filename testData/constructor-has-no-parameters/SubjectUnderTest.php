<?php

namespace App;

class SubjectUnderTest
{
    public function __construct()
    {
    }
}
