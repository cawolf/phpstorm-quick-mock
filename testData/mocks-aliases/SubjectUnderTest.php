<?php

namespace App;

use Psr\Log\LoggerInterface;
use App\LoggerInterface as MainLoggerInterface;

class SubjectUnderTest
{
    private $logger;
    private $mainLogger;

    public function __construct(LoggerInterface $logger, MainLoggerInterface $mainLogger)
    {
        $this->logger = $logger;
        $this->mainLogger = $mainLogger;
    }
}