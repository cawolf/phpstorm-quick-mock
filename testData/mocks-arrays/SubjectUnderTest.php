<?php

namespace App;

class SubjectUnderTest
{
    private $data;

    public function __construct(array $data)
    {
        $this->data = $data;
    }
}