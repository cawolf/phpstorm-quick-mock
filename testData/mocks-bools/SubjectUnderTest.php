<?php

namespace App;

class SubjectUnderTest
{
    private $data;

    public function __construct(bool $data)
    {
        $this->data = $data;
    }
}