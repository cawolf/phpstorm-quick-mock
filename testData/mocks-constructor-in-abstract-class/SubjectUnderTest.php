<?php

namespace App;

class SubjectUnderTest extends AbstractClass
{
}
