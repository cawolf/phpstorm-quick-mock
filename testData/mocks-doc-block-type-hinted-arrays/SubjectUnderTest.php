<?php

use App\Gitlab\Model\GitlabMapping;
use Http\Client\HttpClient;
use JMS\Serializer\SerializerInterface;

class SubjectUnderTest
{
    private $gitlabClient;
    private $serializer;
    private $mapper;
    private $jiraConfig;

    /**
     * @param string[] $jiraConfig
     */
    public function __construct(HttpClient $gitlabClient, SerializerInterface $serializer, GitlabMapping $mapper, array $jiraConfig)
    {
        $this->gitlabClient = $gitlabClient;
        $this->serializer = $serializer;
        $this->mapper = $mapper;
        $this->jiraConfig = $jiraConfig;
    }
}
