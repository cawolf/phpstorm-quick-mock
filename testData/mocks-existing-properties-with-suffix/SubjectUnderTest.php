<?php

namespace App;

class SubjectUnderTest
{
    private $data;

    public function __construct(string $data)
    {
        $this->data = $data;
    }
}