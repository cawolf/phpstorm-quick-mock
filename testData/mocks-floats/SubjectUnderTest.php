<?php

namespace App;

class SubjectUnderTest
{
    private $data;

    public function __construct(float $data)
    {
        $this->data = $data;
    }
}