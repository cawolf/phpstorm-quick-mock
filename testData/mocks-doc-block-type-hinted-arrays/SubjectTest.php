<?php

namespace App\Tests;

use SubjectUnderTest;

class SubjectTest
{
    /** @var SubjectUnderTest */
    private $subject;

    protected function setUp(): void
    {
        $this->subject = new SubjectUnderTest<caret>();
    }
}