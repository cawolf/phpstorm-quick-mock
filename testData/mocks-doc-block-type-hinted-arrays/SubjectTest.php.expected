<?php

namespace App\Tests;

use App\Gitlab\Model\GitlabMapping;
use Http\Client\HttpClient;
use JMS\Serializer\SerializerInterface;
use Prophecy\Prophecy\ObjectProphecy;
use SubjectUnderTest;

class SubjectTest
{
    /** @var HttpClient|ObjectProphecy */
    private $gitlabClient;
    /** @var SerializerInterface|ObjectProphecy */
    private $serializer;
    /** @var GitlabMapping|ObjectProphecy */
    private $mapper;
    /** @var array */
    private $jiraConfig;
    /** @var SubjectUnderTest */
    private $subject;

    protected function setUp(): void
    {
        $this->gitlabClient = $this->prophesize(HttpClient::class);
        $this->serializer = $this->prophesize(SerializerInterface::class);
        $this->mapper = $this->prophesize(GitlabMapping::class);
        $this->jiraConfig = [];

        $this->subject = new SubjectUnderTest($this->gitlabClient->reveal(), $this->serializer->reveal(), $this->mapper->reveal(), $this->jiraConfig);
    }
}